WordPress/Classes/Utility/Media/MediaAssetExporter.swift

```diff
@@ -0,0 +1,236 @@
+            let onExportSessionCompletion: (AVAssetExportSession) -> Void = { (session) in
+                // Guard that the session completed, or return an error.
+                guard session.status == .completed else {
+                    if let error = session.error {
+                        onError(self.exporterErrorWith(error: error))
+                    } else {
+                        onError(AssetExportError.failedExportingVideoDuringExportSession)
+                    }
+                    return
+                }
+                // Finally complete with the export URL.
+                onCompletion(AssetExport.exportedVideo(MediaVideoExport(url: exportURL,
+                                                                        fileSize: exportURL.resourceFileSize,
+                                                                        duration: session.asset.duration.seconds)))
+            }
+            let options = PHVideoRequestOptions()
+            options.isNetworkAccessAllowed = true
+            let manager = PHImageManager.default()
+            // Begin export by requesting an export session.
+            manager.requestExportSession(forVideo: asset,
+                                         options: options,
+                                         exportPreset: AVAssetExportPresetHighestQuality,
+                                         resultHandler: { (session, info) -> Void in
+                                            guard let session = session else {
+                                                onExportSessionError(info?[PHImageErrorKey] as? Error)
+                                                return
+                                            }
+                                            // Configure the export session.
+                                            session.shouldOptimizeForNetworkUse = true
+                                            session.outputURL = exportURL
+                                            session.outputFileType = videoResource.uniformTypeIdentifier
+                                            // Trigger the export with the completion handler.
+                                            session.exportAsynchronously {
+                                                onExportSessionCompletion(session)
+                                            }
+            })
+        } catch {
+            onError(exporterErrorWith(error: error))
+        }
+    }
+
+    /// Exports and writes an asset's GIF data to a local Media URL.
+    ///
+    /// - parameter onCompletion: Called on successful export, with the local file URL of the exported asset.
+    /// - parameter onError: Called if an error was encountered during export.
+    ///
+    fileprivate func exportGIF(forAsset asset: PHAsset, resource: PHAssetResource, onCompletion: @escaping (AssetExport) -> Void, onError: @escaping (MediaExportError) -> Void) {
```

Comment:
For this one, the goal is to have a very explicit method for handling a GIF export, which is detected via the generic image export above via `fileprivate func exportImage(forAsset asset:onCompletion:onError:)`.

While it does in fact simply export the raw data itself from the `PHAsset`, there's not really other cases in which we would be exporting the raw data, and _not_ resizing the image through the `MediaImageExporter`. We could slightly refactor to reuse some of the code, though, perhaps a bit later.